sbt-plugin/src/main/scala/akka/grpc/sbt/AkkaGrpcPlugin.scala

```diff
@@ -69,9 +81,53 @@ object AkkaGrpcPlugin extends AutoPlugin {
           unmanagedResources := { Defaults.collectFiles(unmanagedResourceDirectories, includeFilter, excludeFilter).value },
           resources := managedResources.value ++ unmanagedResources.value)))
 
-  def adaptAkkaGenerator(targetPath: File)(generatorAndSettings: GeneratorAndSettings): Target = {
-    val adapted = new ProtocBridgeSbtPluginCodeGenerator(generatorAndSettings.generator)
-    val generator = JvmGenerator(generatorAndSettings.generator.name, adapted)
-    (generator, generatorAndSettings.settings) -> targetPath
+  private def targetsFor(targetPath: File, defaultSettings: Seq[String], stubs: Seq[AkkaGrpc.TargetStub], languages: Seq[AkkaGrpc.TargetLanguage]): Seq[protocbridge.Target] = {
+    val generators = generatorsFor(stubs, languages, defaultSettings)
+    // TODO no way to provide per language + stub settings/target dirs - not sure if needed
+    generators.map { case (generator, settings) => protocbridge.Target(generator, targetPath, settings) }
+  }
+
+  // creates a seq of generator and per generator settings
+  private def generatorsFor(stubs: Seq[AkkaGrpc.TargetStub], languages: Seq[AkkaGrpc.TargetLanguage], defaultSettings: Seq[String]): Seq[(protocbridge.Generator, Seq[String])] = {
+    // these two are the model/message (protoc) generators
+    def ScalaGenerator: (protocbridge.Generator, Seq[String]) = (protocbridge.JvmGenerator("scala", ScalaPbCodeGenerator), defaultSettings)
+    // we have a default flat_package, but that doesn't play with the java generator (it fails)
+    def JavaGenerator: (protocbridge.Generator, Seq[String]) = (PB.gens.java, defaultSettings.filterNot(_ == "flat_package"))
```

Comment:
I think we should filter this 'the other way around', and only pass through options that the Java generator supports. I think we do that somewhere but can't find it right now...